iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -170,5 +170,10 @@ private static string CreateAmqpCbsAudience(IConnectionCredentials connectionCre
             // SAS tokens granted to a group SAS authenticated client will scoped to the IoT hub-level; for example, myHub.azure-devices.net
             return connectionCredentials.HostName;
         }
+
+        public void Dispose()
+        {
+            throw new NotImplementedException();
```

Comment:
Same comment here. You need to dispose the _loopCancellationTokenSource instance here, not throw NotImplementedException